astronomer/providers/apache/hive/hooks/hive.py

```diff
@@ -48,3 +49,37 @@ async def partition_exists(self, table: str, schema: str, partition: str, pollin
         if len(results) == 0:
             return "failure"
         return "success"
+
+    @staticmethod
+    def parse_partition_name(partition: str) -> Tuple[str, str, str]:
+        """Parse partition string into schema, table, and partition."""
+        first_split = partition.split(".", 1)
+        if len(first_split) == 1:
+            schema = "default"
+            table_partition = max(first_split)  # poor man first
+        else:
+            schema, table_partition = first_split
+        second_split = table_partition.split("/", 1)
+        if len(second_split) == 1:
+            raise ValueError("Could not parse " + partition + "into table, partition")
+        else:
+            table, partition = second_split
+        return schema, table, partition
+
+    def check_partition_exists(self, schema: str, table: str, partition: str) -> bool:
+        """
+        Check whether given partition exist or not.
+
+        :param schema: Name of the Hive schema.
+        :param table: Name of the table.
+        :param partition: Name of the partition
+        """
+        self.log.info("Poking for %s.%s/%s", schema, table, partition)
```

Comment:
This isn't poking thought, as in the function itself does a one-off operation

Reply:
yes, changed the log message